frontend/src/v5/ui/routes/viewer/viewer.tsx

```diff
@@ -56,6 +56,15 @@ export const Viewer = () => {
 		return () => ViewerService.off(VIEWER_EVENTS.CLICK_PIN, handlePinClick);
 	}, [tickets]);
 
+	useEffect(() => {
+		TeamspacesActionsDispatchers.setCurrentTeamspace(teamspace);
+	}, [teamspace]);
+
+	useEffect(() => {
+		ProjectsActionsDispatchers.setCurrentProject(project);
+		ProjectsActionsDispatchers.fetch(teamspace);
```

Comment:
shouldn't this go in the useEffect above?
Afterall you want to fetch the projects when the teamspace changes as opposed to when the project does.

Also, dashboardViewerLayout does something similar, but it does it only once (note the dependencies list is empty). Could it be that replacing its useEffect code with what you have here is enough?